BepInEx.IL2CPP/Preloader.cs

```diff
@@ -83,5 +86,48 @@ public static void Run()
                 throw;
             }
         }
+
+        private static void InitializeUnityVersion()
+        {
+            try
+            {
+                ConfigEntry<String> config = ConfigFile.CoreConfig.Bind(
+                    "Unity", "Version",
+                    String.Empty,
+                    "Allows you to manually set the Unity version .");
+
+                String version = config.Value;
+                if (String.IsNullOrWhiteSpace(version))
+                {
+                    version = //Version.Parse(Application.unityVersion);
+                        Process.GetCurrentProcess().MainModule.FileVersionInfo.FileVersion;
+
+                    PreloaderLogger.Log.LogDebug($"Unity version obtained from main application module: [{version}]");
+                }
+                else
+                {
+                    PreloaderLogger.Log.LogDebug($"Unity version obtained from the config: [{version}]");
+                }
+
+                String[] parts = version.Split('.');
+                Int32 major = 0;
+                Int32 minor = 0;
+                Int32 build = 0;
+
+                // Issue #229 - Don't use Version.Parse("2019.4.16.14703470L&ProductVersion")
+                major = Int32.Parse(parts[0], NumberStyles.Integer, CultureInfo.InvariantCulture);
+                if (parts.Length > 1)
+                    minor = Int32.Parse(parts[1], NumberStyles.Integer, CultureInfo.InvariantCulture);
+                if (parts.Length > 2)
+                    build = Int32.Parse(parts[2], NumberStyles.Integer, CultureInfo.InvariantCulture);
```

Comment:
What if the user mistakingly gives a non-integer? That will cause a fail that will not initialize Unhollower properly. I'd suggest either using `int.TryParse` or better yet, define a RegEx here instead.
A RegEx to me looks like a better solution here overall as it will both check the validity and will allow extracting the version number from more complex strings.

Reply:
I think if the user was typing wrong, we should crash with an error, shouldn't we?

Reply:
If the user types wrong, I'd rather warn and initialize Unhollower with its default Unity version-specific handlers.
I'd err on the side of letting Unhollower decide the version handler to use if no Unity version is given, since Unhollower itself has that ability.

If the value is wrong, I'd consider resetting it like we do for other config options.

Reply:
Done.
But I didn't use RegEx - the optional groups (2019 / 2019.4 / 2019.4.16 / 2019.4.16.1) only add complexity to the code, and don't make it easier to read.